app/queries/my_facilities/blood_pressure_control_query.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+class MyFacilities::BloodPressureControlQuery
+  include QuarterHelper
+  include MonthHelper
+
+  def initialize(period: :quarter,
+                 quarter: quarter(Time.current),
+                 month: Time.current.month,
+                 year: Time.current.year,
+                 facilities: Facility.all)
+    @period = period
+    @month = month
+    @quarter = quarter
+    @year = year
+    @facilities = facilities
+  end
+
+  def cohort_registrations
+    @period == :month ? monthly_registrations : quarterly_registrations
+  end
+
+  def cohort_controlled_bps
+    @period == :month ? monthly_controlled_bps : quarterly_controlled_bps
+  end
+
+  def cohort_uncontrolled_bps
+    @period == :month ? monthly_uncontrolled_bps : quarterly_uncontrolled_bps
+  end
+
+  def all_time_bps
+    LatestBloodPressuresPerPatient
+      .where(bp_facility_id: @facilities)
+  end
+
+  def all_time_controlled_bps
+    all_time_bps
+      .where('bp_recorded_at > ?', Date.current - 90.days)
+      .where('systolic < 140 AND diastolic < 90')
+  end
+
+  private
+
+  def quarterly_registrations
+    patients = Patient.where(registration_facility: @facilities)
+    previous_cohort = previous_year_and_quarter(@year, @quarter)
+
+    patients.where('recorded_at > ? AND recorded_at <= ?',
+                   quarter_start(*previous_cohort),
+                   quarter_end(*previous_cohort))
+  end
+
+  def quarterly_bps
+    cohort_registrations = quarterly_registrations
+    LatestBloodPressuresPerPatientPerQuarter
+      .where(patient: cohort_registrations)
+      .where(year: @year, quarter: @quarter)
+  end
+
+  def quarterly_controlled_bps
+    quarterly_bps.where('systolic < 140 AND diastolic < 90')
+  end
+
+  def quarterly_uncontrolled_bps
+    quarterly_bps.where('systolic >= 140 OR diastolic >= 90')
+  end
+
+  def monthly_registrations
+    patients = Patient.where(registration_facility: @facilities)
```

Comment:
Not sure if this helps much, but why would we do this for `facilities` and not for the other variables we're instantiating?

Reply:
Because we access this one in several places, it's idiomatic Ruby practice to access through a method. The other vars being instantiated are also never accessed directly, but only through the query methods in this class 😉 

Not a huge deal, I'll let you address/resolve as you see fit 👍